app/controllers/application_controller/indexes.rb

```diff
@@ -1,48 +1,170 @@
 # frozen_string_literal: true
 
+##############################################################################
+#
+#  :section: Indexes
+#
+##############################################################################
 # see application_controller.rb
+# rubocop:disable Metrics/ModuleLength
 module ApplicationController::Indexes
   def self.included(base)
     base.helper_method(:paginate_numbers)
-    base.extend(ClassMethods)
   end
 
-  module ClassMethods
-    def index_subaction_param_keys
-      @index_subaction_param_keys ||= []
+  # Get args from a param subaction, or if none given, the unfiltered_index.
+  def build_index_with_query
+    bail = false
+    index_active_params.each do |subaction|
+      next if params[subaction].blank?
+
+      query, display_opts = send(index_param_method_or_default(subaction))
+      # Some actions may redirect instead of returning a query.
+      # If we have the param but get a blank query we should bail.
+      if query.blank?
+        bail = true
+        break
+      end
+      # If we have a query, display it.
+      return filtered_index(query, display_opts)
     end
+    return if bail
+
```

Comment:
```suggestion
    index_active_params.each do |subaction|
      next if params[subaction].blank?

      query, display_opts = send(index_param_method_or_default(subaction))
      # Some actions may redirect instead of returning a query.
      # If we have the param but get a blank query we should bail.
      return if query.blank?

      # If we have a query, display it.
      return filtered_index(query, display_opts)
    end
```

Reply:
Ah - thanks for the suggestion. That's what I had originally, but Rubocop `Lint/NonLocalExitFromIterator` didn't like it. It said "Non-local exit from iterator, without return value. `next`, `break`, `Array#find`, `Array#any?`, etc. is preferred."

I tried switching it to `break`, but then the tests fail with "AbstractController::DoubleRenderError". That's when i tried the `bail` thing. But I'm just now tweaking your version to explicitly return `nil` and that's working, and not offending Rubocop. 

Committing that now.  Thank you! The `bail` was bugging me.